include/subsystems/odometry/odometry_nwheel.h

```diff
@@ -0,0 +1,255 @@
+ * calculate a pose delta, then integrate the deltas over time to determine the robot's position.
+ *
+ * This is a "set and forget" class, meaning once the object is created, the robot will immediately begin
+ * tracking it's movement in the background.
+ *
+ * https://rit.enterprise.slack.com/files/U04112Y5RB6/F080M01KPA5/predictperpindiculars2.pdf
+ * 2024-2025 Notebook: Entries/Software Entries/Localization/N-Pod Odometry
+ *
+ * @author Jack Cammarata, Richie Sommers
+ * @date Nov 14 2024
+ *
+ */
+
+/**
+ * tracking_wheel_cfg_t holds all the specifications for a single tracking wheel
+ * The units for x, y, and radius will determine the units of the position estimate
+ */
+typedef struct {
+  double x;         /**< x position of the center of the wheel */
+  double y;         /**< y position of the center of the wheel */
+  double theta_rad; /**< angle between wheel direction and x axis in the robot frame */
+  double radius;    /**< radius of the wheel */
+} tracking_wheel_cfg_t;
+
+template <int WHEELS> class OdometryNWheel : public OdometryBase {
+public:
+  /**
+   * Construct a new Odometry N Wheel object
+   *
+   * @param encoders std::array containing each CustomEncoder
+   * @param wheel_configs std::array containing each tracking_wheel_cfg
+   * @param is_async true to constantly run in the background
+   */
+  OdometryNWheel(
+    const std::array<CustomEncoder, WHEELS> &encoders, const std::array<tracking_wheel_cfg_t, WHEELS> wheel_configs,
+    vex::inertial *imu, bool is_async
+  )
+      : OdometryBase(is_async), imu(imu), encoders(encoders) {
+    Eigen::Matrix<double, WHEELS, 3> transfer_matrix;
+    for (int i = 0; i < WHEELS; i++) {
+      double x = wheel_configs[i].x;
+      double y = wheel_configs[i].y;
+      double theta_rad = wheel_configs[i].theta_rad;
+      double radius = wheel_configs[i].radius;
+      wheel_radii(i) = radius;
+
+      double x_factor = cos(theta_rad);
+      double y_factor = -sin(theta_rad);
+      double theta_factor = -(x * sin(theta_rad)) - (y * cos(theta_rad));
+
+      if (abs(y_factor) < 0.000000001) {
+        y_factor = 0;
+      }
+      if (abs(x_factor) < 0.000000001) {
+        x_factor = 0;
+      }
+
+      transfer_matrix.row(i) << x_factor, y_factor, theta_factor;
+    }
+
+    transfer_matrix_pseudoinverse = transfer_matrix.completeOrthogonalDecomposition().pseudoInverse();
+
+    old_heading = 0;
+  }
+
+  /**
+   * Update the current position of the robot once, using the current state of
+   * the encoders and the previous known location
+   *
+   * @return the robot's updated position
+   */
+  pose_t update() override {
+    Eigen::Vector<double, WHEELS> radian_deltas;
+
+    for (int i = 0; i < WHEELS; i++) {
+      double angle_rad = encoders[i].position(rev) * M_PI * 2;
+      radian_deltas(i) = angle_rad - old_wheel_angles[i];
+
+      old_wheel_angles[i] = angle_rad;
+    }
+
+    pose_t updated_pos = calculate_new_pos(radian_deltas, current_pos);
+
+    angle = 0;
```

Comment:
anlge is set to 0 here then immediately changed in the next line. can probably get rid of it here